### PR TITLE
#34. Add 'No recent actions' stub.

### PR DESCRIPTION
**Pull request by bogdan-evtushenko**

### CODE REVIEW

File: app/src/main/java/com/bogdan/codeforceswatcher/features/actions/ActionsAdapter.kt [hunk 1]
```diff
@@ -16,36 +15,50 @@ import kotlinx.android.synthetic.main.view_action_item.view.*
 class ActionsAdapter(
     private val context: Context,
     private val itemClickListener: (Int) -> Unit
-) : RecyclerView.Adapter<ActionsAdapter.ViewHolder>() {
+) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
```
Comment (yevhenii-kanivets) [inline app/src/main/java/com/bogdan/codeforceswatcher/features/actions/ActionsAdapter.kt]: Good catch 👍

File: app/src/main/java/com/bogdan/codeforceswatcher/features/actions/ActionsAdapter.kt [hunk 2]
```diff
@@ -16,36 +15,50 @@ import kotlinx.android.synthetic.main.view_action_item.view.*
...
 
     private var items: List<ActionItem> = listOf()
 
     override fun getItemCount() = items.size
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
-        val layoutInflater = LayoutInflater.from(context).inflate(R.layout.view_action_item, parent, false)
-        return ViewHolder(layoutInflater, itemClickListener)
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
+        val layoutInflater = when (viewType) {
```
Comment (yevhenii-kanivets) [inline app/src/main/java/com/bogdan/codeforceswatcher/features/actions/ActionsAdapter.kt]: It's a view returned here, not a `layoutInflater`.

File: app/src/main/java/com/bogdan/codeforceswatcher/features/actions/ActionsAdapter.kt [hunk 3]
```diff
@@ -16,36 +15,50 @@ import kotlinx.android.synthetic.main.view_action_item.view.*
...
+            STUB_VIEW_TYPE -> LayoutInflater.from(context).inflate(R.layout.view_action_stub, parent, false)
+            else -> LayoutInflater.from(context).inflate(R.layout.view_action_item, parent, false)
+        }
+        return when (viewType) {
```
Comment (yevhenii-kanivets) [inline app/src/main/java/com/bogdan/codeforceswatcher/features/actions/ActionsAdapter.kt]: And you have a duplicated `when`, which a CC smell (Clean Code Smell = запашок описанный в книге Clean Code).

File: app/src/main/java/com/bogdan/codeforceswatcher/features/actions/ActionsFragment.kt [hunk 4]
```diff
@@ -36,7 +36,8 @@ class ActionsFragment : Fragment(), SwipeRefreshLayout.OnRefreshListener,
 
     override fun newState(state: ActionsState) {
         swipeRefreshLayout.isRefreshing = (state.status == ActionsState.Status.PENDING)
-        actionsAdapter.setItems(state.actions.map { ActionItem(it) })
+        if (state.actions.isEmpty()) actionsAdapter.setItems(listOf(ActionItem.Stub))
```
Comment (yevhenii-kanivets) [inline app/src/main/java/com/bogdan/codeforceswatcher/features/actions/ActionsFragment.kt]: I think, it's better extract `items` and use `if` there.

File: app/src/main/java/com/bogdan/codeforceswatcher/features/actions/models/Action.kt [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+import android.text.SpannableStringBuilder
+import com.bogdan.codeforceswatcher.CwApp
+import com.bogdan.codeforceswatcher.R
+import com.bogdan.codeforceswatcher.features.users.models.colorTextByUserRank
+import org.ocpsoft.prettytime.PrettyTime
+import java.util.*
+
+sealed class ActionItem {
+    class Action(action: CFAction) : ActionItem() {
```
Comment (yevhenii-kanivets) [inline app/src/main/java/com/bogdan/codeforceswatcher/features/actions/models/Action.kt]: Add an empty line before, please :)

File: app/src/main/res/values-ru-rRU/strings.xml [hunk 6]
```diff
@@ -56,5 +56,5 @@ Codeforces Watcher позволяет:
 5. Задачи также будут доступны в ближайшем будущем…
 
 Загружайте прямо сейчас на Google Play - https://play.google.com/store/apps/details?id=com.bogdan.codeforceswatcher"</string>
-
+    <string name="actions_on_the_way">Последние действия на пути к вашему устройству…</string>
```
Comment (yevhenii-kanivets) [inline app/src/main/res/values-ru-rRU/strings.xml]: Ох же и звучит) Давай лучше "Свежайшие новости "

File: app/src/main/res/values-uk-rUA/strings.xml [hunk 7]
```diff
@@ -56,4 +56,5 @@ Codeforces Watcher дозволяє:
 5. Задачі також стануть доступні у найближчому майбутньому…
 
 Завантажуйте прямо зараз на Google Play - https://play.google.com/store/apps/details?id=com.bogdan.codeforceswatcher"</string>
+    <string name="actions_on_the_way">Останні дії на шляху до вашого пристрою…</string>
```
Comment (yevhenii-kanivets) [inline app/src/main/res/values-uk-rUA/strings.xml]: свіжі новини

File: app/src/main/java/com/bogdan/codeforceswatcher/features/actions/ActionsAdapter.kt (hunk 3 above)
Comment (bogdan-evtushenko) [reply app/src/main/java/com/bogdan/codeforceswatcher/features/actions/ActionsAdapter.kt]: For some reason I thought it was not quite right) But it seemed to me smaller in the lines of code

**Pull request merged**